app/routes/signup.tsx

```diff
@@ -135,61 +142,198 @@ export const loader: LoaderFunction = async ({request}) => {
   })
 }
 
+const TEAM_MAP = {
+  BLUE: {
+    image: images.alexBlue,
+    label: 'Blue Team',
+    focusClassName: 'ring-team-blue',
+  },
+  RED: {
+    image: images.alexRed,
+    label: 'Red Team',
+    focusClassName: 'ring-team-red',
+  },
+  YELLOW: {
+    image: images.alexYellow,
+    label: 'Yellow Team',
+    focusClassName: 'ring-team-yellow',
+  },
+}
+
+interface TeamOptionProps {
+  team: keyof typeof TEAM_MAP
+  error?: string | null
+  selected: boolean
+}
+
+function TeamOption({team: value, error, selected}: TeamOptionProps) {
+  const team = TEAM_MAP[value]
+
+  return (
+    <div
+      className={clsx(
+        'relative col-span-full mb-3 bg-gray-100 dark:bg-gray-800 rounded-lg focus-within:outline-none ring-team-current ring-offset-4 focus-within:ring-2 lg:col-span-4 lg:mb-0',
+        team.focusClassName,
+        {
+          'ring-2 ring-offset-team-current': selected,
+          'dark:ring-offset-gray-900 ring-offset-white': !selected,
+        },
+      )}
+    >
+      {selected ? (
+        <span className="absolute left-9 top-9 text-team-current">
+          <CheckIcon />
+        </span>
+      ) : null}
+
+      <label className="block pb-12 pt-20 px-12 text-center">
+        <input
+          className="sr-only"
+          type="radio"
+          name="team"
+          value={value}
+          aria-describedby={error ? 'team-error' : undefined}
+        />
+        <img className="block mb-16" src={team.image.src} alt="" />
+        <H6>{team.label}</H6>
+      </label>
+    </div>
+  )
+}
+
 export default function NewAccount() {
   const data = useRouteData<LoaderData>()
+  const [formValues, setFormValues] = React.useState({
+    firstName: '',
+    team: '',
+  })
+
+  const formIsValid =
+    formValues.firstName.trim().length > 0 &&
+    Object.keys(TEAM_MAP).includes(formValues.team)
+
   return (
-    <div>
-      <h2>Welcome to kentcdodds.com</h2>
-      <p>To create your account for {data.email}, choose your team:</p>
-      <Form method="post" noValidate>
-        <div>
-          <label htmlFor="firstName">First Name: </label>
-          <input
-            name="firstName"
-            id="firstName"
-            required
-            defaultValue={data.fields?.firstName ?? ''}
-          />
-          {data.errors?.firstName ? (
-            <div
-              role="alert"
-              className="dark:text-red-300 text-red-800"
-              id="body-error"
-            >
-              {data.errors.firstName}
+    <div className="mt-12">
+      <style>{`
+          :root {
+            --color-team-current: var(--color-team-${(
+              formValues.team || 'UNKNOWN'
+            ).toLowerCase()}); 
+          }
+        `}</style>
```

Comment:
Oh, clever idea! Let's add a comment explaining what this is doing and why we can't use the `--color-team-current` that's defined in the `root.tsx` in case I come back to this in a few months and I forget 😅

Reply:
On second thought. Do you think that we could use a `useEffect` instead that does this: 

```tsx
document.documentElement.style.setProperty('--color-team-unknown', `var(--color-team-${(formValues.team || 'UNKNOWN').toLowerCase()})`)
```

That way the circle in the top right updates too:

![image](https://user-images.githubusercontent.com/1500684/124310094-ef021d00-db28-11eb-9544-6250965f9756.png)

And actually it just occurred to me that we know enough by this time to get the user's avatar as well, so I'll work on that bit.

Reply:
Starting to think I wanna put this in context so the avatar can be updated too...

Reply:
Go ahead and leave this as-is. I'll come back around to it.